webpack/webpack-env.d.ts

```diff
@@ -101,3 +102,137 @@ declare var __non_webpack_require__: any;
  * Equals the config option debug
  */
 declare var DEBUG: boolean;
+
+/**
+ * Webpack Hot Module Replacement
+ * https://webpack.github.io/docs/hot-module-replacement.html
+ */
+declare namespace __WebpackHotModuleReplacement {
+  interface Module {
```

Comment:
Thank you for your review :smile:

> I'm not sure which properties are actually supported

Neither do I.  So far I just defined webpack's `module` from my running node environment.  But for example I omitted the `paths` prop since it's present both in 'bare' `module` and webpack-configured `module` but not defined in `node.d.ts`.  I didn't want bring in irrelevant issues so that was simply copy-and-pasted.

> Accounting for the various configurations could be done with a union type, but that wouldn't be very practical

I agree with that.  Casting `module` manually might lose the beauty of static typing.  It could be possible to prepare separate definition file per configuration like you said in your comments https://github.com/borisyankov/DefinitelyTyped/pull/6558#issuecomment-153052463.  This could be done in future by someone who needs it.  But so far the mininum definition will suffice and is better than using `any`.

> A few minor/cosmetic remarks

I see.  Thanks I'll move `Module` into `__WebpackModuleApi`